orchestrator-salt/src/main/resources/salt-common/salt/fluent/template/check_fluent_plugins.sh.j2

```diff
@@ -60,6 +60,7 @@ function check_and_install_plugin() {
 }
 
 function main() {
+{# Protect this via a grains[prewarmed] check}
```

Comment:
and how to check the plugin versions are right for the image (older pre-warmed) or CB version? in these cases we want to fail during deploy (if no network) or upgrade to the right versions for the plugins (if you have network)

Reply:
How often are the plugin versions changed? When did we change them last. Also, how do we know the latest version which can be used - does Cloudbreak know this?, is the change made manually, is it just the latest version of the plugin that exists in the wild?
Either way - I think prewarmed images + some kind of grains could be used to skip the check.
Oliver, I'll reach out to you offline about the td-agent package - which does slow down bootup.

Reply:
for instance it changed yesterday. And yes also CB knows about the change, if it should be upgraded (because of CB) and there is an old image that we are using. In case of no network we should see as an error message to not hide that something won't functioning on the node.

There are 2 main use cases with version conflict:
1. if we have new image in production but a new CB is not deployed - version conflict (plugins)
- new version in the prewarmed image
- CB still uses older one (in the salt scripts)
-> outcome: here we are checking ">=", so it won't upgrade anything
2. New CB deployed, a node uses an older image (it happens often) - version conflict (plugins)
- old version in the prewarmed image
- CB requires a new plugin version (based on the salt definitions) because of some new features or fixes
outcomes:
a.) if there is network and you can access cloudera or non-cloudera gem repo -> update gems
b.) if there is no network -> install fails and tell the user they need to use a new image, update gems manually or do changes in the firewall in order to update the plugins

If we skip the second use case (because prewarm check), customer will see errors a bit later during the installation, which can be much more complicated to figure out at first (like no logs on cloud-storage or a service fails etc.)